src/color.cc

```diff
@@ -727,11 +724,11 @@ rgba_from_hex_string(const char *str, short *ok) {
 
 static int32_t
 rgba_from_name_string(const char *str, short *ok) {
-  int i = 0;
-  struct named_color color;
-  while ((color = named_colors[i++]).name) {
-    if (*str == *color.name && 0 == strcmp(str, color.name))
-      return *ok = 1, color.val;
+  std::string lowered(str);
+  std::transform(lowered.begin(), lowered.end(), lowered.begin(), tolower);
+  auto color = named_colors.find(lowered);
+  if (color != named_colors.end()) {
+    return *ok = 1, color->second;
```

Comment:
I'm not sure which C++ standard we are targeting, but this function could return `std::optional<int32_t>` and drop the `ok` parameter if we target C++17

Reply:
That'd be nice. [Node targets C++14](https://github.com/nodejs/node/blob/573744caaedf2f941e152b8e9533cd5629d55c1c/common.gypi#L348) currently though, so I'm afraid to target past that.